index.rst

```diff
@@ -1,29 +1,481 @@
+   * - ``dimension_data``
+     - single-block
+     - | All butler dimension records referenced by any data ID in the graph.
+       | While this nominally grows with the size of the full graph, when all
+       | dimensions are stored (and compressed) together, the total size is
+       | in practice always small compared to other components.
+   * - ``thin_quanta``
+     - single-block
+     - Task labels and data IDs (only) for all quanta.
+   * - ``init_quanta``
+     - single-block
+     - | Init-input and init-output dataset information, represented as one
+       | special "init quantum" for each task, using the same model as
+       | ``full_quanta``.
+   * - ``full_quanta``
+     - multi-block
+     - | Input and output dataset information, including data IDs, UUIDs,
+       | and datastore records for each quantum.
+   * - ``quantum_addresses``
+     - address (sorted by UUID)
+     - | UUID to integer ID mapping for all quanta, byte offsets and sizes
+       | for ``full_quanta``.
+
+A deep read of just a few quanta (identified by UUID) would thus involve the following steps:
+
+1. Read the ``header`` to verify file format versions and load general metadata.
+2. Read the complete ``pipeline_graph``, ``dimension_data`` and ``init_quanta`` components.
+3. Search the ``quantum_addresses`` file for the byte offsets and integer IDs for the UUIDs of interest.
+4. Read the blocks of the ``full_quanta`` component that according to those byte offsets.
+5. Use a :class:`~lsst.daf.butler.DimensionDataAttacher` to attach dimension records to all loaded quantum and dataset data IDs.
+
+A shallow read of all quanta (e.g. for ``bps transform``) would read different components:
+
+1. Read the ``header`` to verify file format versions and load general metadata.
+2. Read the complete ``pipeline_graph``, ``quantum_edges``, ``thin_quanta``, ``init_quanta``, and ``quantum_addresses`` component.
+3. Transform the internal IDs to UUIDs, and use the edge list to make a :class:`networkx.DiGraph` of all quanta.
+
+.. _wire-graphs:
+
+Wire Quantum Graphs
+-------------------
+
+A "wire" quantum graph is a lightweight version of the predicted graph that can be sent from prompt-processing worker pods to its ingestion service efficiently.
+It does not need to support :class:`~lsst.daf.butler.QuantumBackedButler` and can assume its reader has access to a full butler.
+
+The exact components in a wire quantum graph are still TBD, as they depend on how much information the Prompt Processing ingest service gets from other sources (e.g. does it already know the exact :class:`~pipeline_graph.PipelineGraph` the worker pods are running).
+Because prompt-processing graphs only have a single quantum for each task (and rarely more than one dataset of each type), the expensive components are quite different from the expensive components in a large batch graph, and there is no need to support single-quantum partial reads with multi-block components and address files.
```

Comment:
Took me a moment to parse this and I think you are saying: a prompt-processing graph has multiple quanta in it but they are connected in a linear fashion with only one instance of a specific Task.